.github/workflows/deploy-book.yaml

```diff
@@ -59,27 +61,33 @@ jobs:
           unzip book.zip
           rm -f book.zip
 
+      - name: Debug
+        run: pwd
+
+      - name: Debug
+        run: ls -la book/_build/html
+
       - name: Deploy to GitHub Pages
         uses: JamesIves/github-pages-deploy-action@v4
         if: |
           (github.ref == 'refs/heads/main' && inputs.cname == 'None')
         with:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: ${{ inputs.publish_dir }}
-          CLEAN: true
-          KEEP_HISTORY: false
-          BASE_PATH: ${{ inputs.destination_dir }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: gh-pages
+          folder: ${{ inputs.publish_dir }}
+          clean: true
+          clean-exclude: "preview" # keep existing previews from other PRs
+          target-folder: ${{ inputs.destination_dir }}
 
       - name: Deploy to GitHub Pages with custom domain
         uses: JamesIves/github-pages-deploy-action@v4
         if: |
           (github.ref == 'refs/heads/main' && inputs.cname != 'None')
         with:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: ${{ inputs.publish_dir }}
-          CLEAN: true
-          KEEP_HISTORY: false
-          BASE_PATH: ${{ inputs.destination_dir }}
-          CNAME: ${{ inputs.cname }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: gh-pages
+          folder: ${{ inputs.publish_dir }}
+          clean: true
+          clean-exclude: "preview" # keep existing previews from other PRs
+          target-folder: ${{ inputs.destination_dir }}
+          CNAME: ${{ inputs.cname }} # how can we set this for the new action?
```

Comment:
This is still not solved. Does someone have an example deployment that uses this input?

Reply:
Could `git-config-name` be similar to what you are looking for?

Reply:
I am honestly really unsure. Could somebody test this out on a doc that uses this feature?

Reply:
Our Foundations site uses this feature. Currently fixing some broken links in Foundations that are preventing CI tests from passing (https://github.com/ProjectPythia/pythia-foundations/pull/495), once those are merged I'll do some testing here.

Reply:
I did a bit of reading on the CNAME feature, including the docs that @jbusecke linked above. Previously we were passing this argument through to [peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages). As far as I understand, the only effect of this argument is to ensure that a CNAME file is created at the root of the deployment branch (we use `gh-pages`, e.g. [this file in the Foundations repo](https://github.com/ProjectPythia/pythia-foundations/blob/gh-pages/CNAME)).

That file hasn't changed in three years because we've never changed the custom domain name.

The [docs for peaceiris/actions-gh-pages](https://github.com/peaceiris/actions-gh-pages?tab=readme-ov-file#%EF%B8%8F-add-cname-file-cname) also say 
> To add the CNAME file, we can set the cname option. Alternatively, put your CNAME file into your publish_dir. (e.g. public/CNAME)

The [docs for JamesIves/github-pages-deploy-action](https://github.com/JamesIves/github-pages-deploy-action?tab=readme-ov-file#additional-build-files-) also refer to manual commit of the CNAME file into the deployment branch:
> If you're using a custom domain and require a CNAME file, or if you require the use of a .nojekyll file, you can safely commit these files directly into the deployment branch without them being overridden after each deployment

My conclusion is that we should just remove this input argument. Manually creating the CNAME file in the `gh-pages` branch is a one-time step. And within Project Pythia, we only use this feature for the Foundations and Cookbook Gallery repos. It's not something that we are actively supporting for Cookbook contributors.